### PR TITLE
Layout

### PR DESCRIPTION
**Pull request by mqueb**

### CODE REVIEW

File: Mod Configuration Menu/Mods/BG3MCM/ScriptExtender/Lua/Client/IMGUILayer.lua [hunk 1]
```diff
@@ -457,63 +403,4 @@ function IMGUILayer:CreateModMenuSetting(modGroup, setting, modSettings, modGUID
...
-    if not MCM_WINDOW then
-        return
-    end
-
-    local modTabs = self.modsTabs[modGUID].modTabBar
-    local newTab = modTabs:AddTabItem(tabName)
-    newTab.IDContext = modGUID .. "_" .. tabName
-    tabCallback(newTab)
-
-    Ext.Net.PostMessageToServer(Channels.MCM_MOD_TAB_ADDED, Ext.Json.Stringify({
```
Comment (mqueb) [inline Mod Configuration Menu/Mods/BG3MCM/ScriptExtender/Lua/Client/IMGUILayer.lua:515]: that one must be moved i guess

**Comment by AtilioA** [comment]
Thanks for the contribution! I'm having problems with my PC hardware, so I can't work much in this for now. I was able to notice that the vertical scroll (not sure if ScrollY property could change this) was shared between the mod list and a mod's tab, but other than this it's looking great!

**Comment by mqueb** [comment]
i may try to change "container" build in "initFrameLayout"  with somethingelse to avoid scroll problem. 
Mow that layout and Content are separated, it's more simple to try other things

**Comment by mqueb** [comment]
@AtilioA : FYI i have rewrite update visibility to have full support of logical operator. Currently testing

**Comment by AtilioA** [comment]
Current version should be okay, but feel free to open another branch

**Comment by mqueb** [comment]
will wait this one to be merged. Rewrite of visibleIf  finished. 
currently add spcial operation to have visibleIf like:  XXXX ; VisibleIf  yId isVisible true/false
as we havent a complex model, it should provide way to do everything, too.  And it allow to share VisibleIf without rewrite same condition. Will be included with same PR as the rewrite

**Comment by AtilioA** [comment]
I don't think conditions should be THAT complex... wanting to have complex stuff like that sounds bad design on the dependent mod side. You can already chain conditions by depending on value from already conditioned settings

**Comment by AtilioA** [comment]
I noticed `InsertModTab` is currently broken

**Comment by mqueb** [comment]
> I don't think conditions should be _that_ complex... wanting to have very complex conditions sounds like bad design on the dependent mod side. You can already chain conditions by depending on values from already conditioned settings

but an not visible setting still have his value. so it's a way to link condition

**Comment by mqueb** [comment]
> I noticed `InsertModTab` is currently broken

not gat any errors and all is rendered on my tests.   theres isn't any `InsertModTab ` function . Did u mean `InsertModMenuTab` ?

i jsut noticed it was used by IMGUIAPI.  What is the goal of that way to create as, it's all done by IMGLayer:loadMods ?

**Comment by mqueb** [comment]
I also notice that your merge have made `CreateModItem` function come back. i have deleted it as it's refactored to frameManager

**Comment by AtilioA** [comment]
> I notice that your merge have made `CreateModItem` and `AddModTooltip` function come back. i have deleted it as it's refactored to frameManager.
> 
> what u add on `CreateModItem` should go in `FrameManager:CreateMenuButton`  and `CreateModItem` deleted
> u add nothing on `AddModTooltip`  so u can juste remove it

I was just changing the base of the PR so that it is merged to `devel`, feel free to change this stuff

**Comment by AtilioA** [comment]
> > I noticed `InsertModTab` is currently broken
> 
> not got any errors and all is rendered on my tests.   theres isn't any `InsertModTab ` function . Did u mean `InsertModMenuTab`  `CreateModTabBar` `AddTabToModTabBar` that i removed as they arn't used?
> 
> i just noticed it was used by IMGUIAPI.  IMGUIAPI function that called remoed one arn't used
> 
> What is the goal of that way to create as, it's all done by IMGLayer:loadMods ? as i said, not got any problem on my commit.
> May be not compatible with your merge
> 

`InsertModMenuTab` is used by mods to add custom IMGUI logic to MCM, like EasyCheat does

**Comment by mqueb** [comment]
> `InsertModMenuTab` is used by mods to add custom IMGUI rendering inside MCM, like EasyCheat does

will install hat mod to fix it

**Comment by mqueb** [comment]
it's almost fixed. Only wierd thing is that , ezCheat call API before"MCM_WINDOW_READY" is sent ... So tab isn't inialized.

check their code, and call API on there _init ....  and not wait fot MCM windows ready

**Comment by mqueb** [comment]
i think creation of modTab should be mouved inside FrameManager, When the group is created

**Comment by mqueb** [comment]
juste a question about: `Ext.Net.PostMessageToServer(Channels.MCM_MOD_TAB_ADDED, Ext.Json.Stringify({`
it isn't done for modTab from classic blueprint's file.  Should i send it too, or keep only those from API send Message to server ?

**Comment by mqueb** [comment]
have addeed "MCM_MOD_SUBTAB_ACTIVATED"  for  OnActivate Tab created by API. 

FrameLayout have now functions   getGroup (used for add footer)   GetTabBar  (to add new tab).

I juste dont change   IMGUILayer to use `InsertModTab` with his callback  during CreateModMenuFrame//CreateModMenuSubTab

can be done to avoid some ducplication ... but rly not necessary

**Comment by AtilioA** [comment]
> juste a question about: `Ext.Net.PostMessageToServer(Channels.MCM_MOD_TAB_ADDED, Ext.Json.Stringify({`
> it isn't done for modTab from classic blueprint's file.  Should i send it too, or keep only those from API send Message to server ?

Only those from the API

**Comment by mqueb** [comment]
make any change if you want, but it's done i think. I check for other layout allowing scroll fix for menu button.  But i dont find impl of horizontal layout for group inside impl of IMGUI in BG3SE

File: Mod Configuration Menu/Mods/BG3MCM/ScriptExtender/Lua/Client/Components/FrameManager.lua [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+ ---@return any the group Content associated to uuid
+ function FrameManager:GetGroup(uuid)
+     return self.contentGroups[uuid]
+ end
+
+ ---@param uuid string
+ ---@return any the group Content associated to uuid
+ function FrameManager:GetModTabBar(uuid)
+     return self.contentGroups[uuid].Children[1]
+ end
```
Comment (AtilioA) [inline Mod Configuration Menu/Mods/BG3MCM/ScriptExtender/Lua/Client/Components/FrameManager.lua:67]: Apart from not handling nils, the problem with accessing `Children[1]` is that it assumes the first child is the mod tab bar we're interested in. However, the position of elements in a table, if the order of children change or if the desired child is not always at index 1, this approach can lead to incorrect or unexpected behavior, as I have already experience by trying to add stuff to `uiGroupMod`.
I think `contentGroup.Children` should then instead be iterated with a check whether it's the mod tab bar.

**Comment by AtilioA** [comment]
The PR is looking good. I think it just needs tweaking a few things to avoid tech debt and making more clear which mod has been selected (there is no current indication, I think a different color for the button of the current one would be a good first step)

File: Mod Configuration Menu/Mods/BG3MCM/ScriptExtender/Lua/Client/Components/FrameManager.lua (hunk 2 above)
Comment (mqueb) [reply Mod Configuration Menu/Mods/BG3MCM/ScriptExtender/Lua/Client/Components/FrameManager.lua:67]: line 50 a, i create tabBar first always

Comment (AtilioA) [reply Mod Configuration Menu/Mods/BG3MCM/ScriptExtender/Lua/Client/Components/FrameManager.lua:67]: Yes, but that can be broken if that ever changes

**Comment by mqueb** [comment]
i waste so much time on last commit .... hope you're ok with so crappy code ...

**Comment by AtilioA** [comment]
Yeah, spacing is not worth doing on IMGUI at least currently 😆

**Comment by mqueb** [comment]
you have removed button fill menu and be centered ?

**Comment by AtilioA** [comment]
I couldn't make it work here, might depend on the scale of the interface or something, but it just wasn't looking good even if changing the number slightly. But the buttons not being centered is just a minor styling problem (and a hard one to solve)

**Comment by mqueb** [comment]
han't try it on other monitor. was fine on mine

File: .vscode/schema.json [hunk 3]
```diff
@@ -357,6 +357,15 @@
                 "VisibleIf": {
                     "$ref": "#/definitions/VisibilityCondition",
                     "description": "Array of Conditions that determine the visibility of this section. Condition is an object with SettingId, Operator, and ExpectedValue."
+                },
+                "Options": {
```
Comment (mqueb) [inline .vscode/schema.json:361]: why an options lvl and not isCollapsible only ? not sure it's usefull

**Comment by AtilioA** [comment]
Thanks again for the work on the layout, I wouldn't have started working on this so soon

**Pull request merged**